File: pset6/readability.py
import cs50


def main():
    text = cs50.get_string("text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentence = count_sentence(text)
    index = math_index(letters, words, sentence)
    # 計算した結果がindexに格納されているのでそれに応じてレベルを表示
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        # indexをstring型にして表示
        print("Grade " + str(index))


def count_sentence(text):
    count = 0
    for i in range(len(text)):
        # . ! ? が文末に存在するのでこれの数をカウント
        asc = ord(text[i])
        if asc == 46 or asc == 33 or asc == 63:
            count += 1
    return count


def count_words(text):
    count = 0
    for i in range(len(text)):
        # 単語はスペース区切りなのでスペースの数をカウント
        asc = ord(text[i])
        if asc == 32:
            count += 1
    count += 1
    return count


def count_letters(text):
    count = 0
    # アルファベットの数をカウント
    for i in range(len(text)):
        if str.isalpha(text[i]):
            count += 1
    return count


def math_index(leng, words, sentence):
    # 計算式に当てはめて計算
    L = leng / words * 100
    S = sentence / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    # 小数点以下を丸める
    ans = round(index)
    return ans


main()

File: pset2/readability.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>

int count_letters(string count_text);
int count_words();
int count_sentence(string s);
double math_index(int len, int words, int sentence);

int main(void)
{
    //get user imput
    string s = get_string("text: ");
    int str = count_letters(s);
    int words = count_words(s);
    int sentence = count_sentence(s);
    double index = math_index(str, words, sentence);

    //Conditional branch according to the calculated result
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)index);
    }

}

int count_words(string s)
{
    int count = 0;

    //Repeat for the number of characters entered
    for (int i = 0, len = strlen(s); i < len; i++)
    {
        //char to ascii
        int asc = (int)s[i];
        //if it was space
        if (asc == 32)
        {
            count++;
        }
    }
    //The number of words is the number of spaces plus one
    count++;
    return (count);
}

int count_letters(string s)
{
    int count = 0;
    for (int i = 0, len = strlen(s); i < len; i++)
    {
        //If it was alphabet
        if (isalpha(s[i]))
        {
            count++;
        }
    }
    return (count);
}

int count_sentence(string s)
{
    int count = 0;

    for (int i = 0, len = strlen(s); i < len; i++)
    {
        int asc = (int)s[i];
        //if (.) (!) (?)
        if (asc == 46 || asc == 33 || asc == 63)
        {
            count++;
        }
    }
    return (count);
}

double math_index(int len, int words, int sentence)
{
    //math with Coleman-Liau index
    double L = (double)len / (double)words * 100;
    double S = (double)sentence / (double)words * 100;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    double ans = round(index);
    return (ans);
}